File: my-react-app/src/Models/RecipeResponse.ts
export interface RecipeResponse {
    id: number;
    title: string;
    description: string;
    ingredients: string[];
    preparationSteps: string;
    cookingTime: number;
    servings: number;
    dietaryInfo?: string;
    containsGluten: boolean;
    status: string;
    createdAt: string;
    updatedAt: string;
    createdByUsername: string;
    photo?: string;
    categories: string[];
}

File: Recipesdemo/src/main/java/Allrecipes/Recipesdemo/Response/RecipeResponse.java
package Allrecipes.Recipesdemo.Response;

import Allrecipes.Recipesdemo.DTOs.IngredientDto;

import java.util.List;

public class RecipeResponse {
    private Long id;
    private String title;
    private String description;
    private List<IngredientDto> ingredients; // Updated to use IngredientDto    private String preparationSteps;
    private int cookingTime;
    private int servings;
    private String dietaryInfo;
    private String status;
    private String createdByUsername;

    // Getters and Setters
}
